src/umpire/ResourceManager.cpp

```diff
@@ -278,6 +278,27 @@ util::AllocationRecord* ResourceManager::deregisterAllocation(void* ptr)
   return m_allocations.remove(ptr);
 }
 
+void ResourceManager::registerExternalAllocation(
+    void* ptr,
+    size_t size,
+    Platform platform)
+{
+  UMPIRE_LOG(Debug, "(ptr=" << ptr << ", size=" << size << ") with " << this );
+
+  auto record = m_allocations.find(ptr);
```

Comment:
This check only checks for the base pointer.  The entire range should be checked.

Reply:
Agree!

Reply:
Actually, this will find any that falls inside the range `[record->m_ptr, record->m_ptr + record->m_size]`. Does that address your concern?

Reply:
I'm not sure about that.  For example, I think it will completely miss an allocation that is in the range ```[record->m_ptr+N, ...]``` where ```N<=record->m_size```.  If I understand correctly, this would yield an overlapping set of records.

Reply:
It would find it, since when `N <= record->m_size`, it would count as being an offset of the base pointer.